test/common/network/connection_impl_test.cc

```diff
@@ -797,6 +797,38 @@ TEST_P(ConnectionImplTest, BasicWrite) {
   disconnect(true);
 }
 
+// Validate dynamic socket options
+TEST_P(ConnectionImplTest, DynamicSocketOptions) {
```

Comment:
Mind doing one test per getter/setter pair?

Reply:
I will do this change.